grpc/execution/validation.go

```diff
@@ -0,0 +1,64 @@
+package execution
+
+import (
+	sequencerblockv1alpha1 "buf.build/gen/go/astria/sequencerblock-apis/protocolbuffers/go/astria/sequencerblock/v1alpha1"
+	"crypto/sha256"
+	"fmt"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+// if sequencer tx is valid, then a unmarshalled ethereum transaction is returned. if not valid, nil is returned
+func ValidateAndUnmarshalSequencerTx(tx *sequencerblockv1alpha1.RollupData, bridgeAddresses map[string]*params.AstriaBridgeAddressConfig, bridgeAllowedAssetIDs map[[32]byte]struct{}) (*types.Transaction, error) {
+	if deposit := tx.GetDeposit(); deposit != nil {
+		bridgeAddress := string(deposit.BridgeAddress.GetInner())
+		bac, ok := bridgeAddresses[bridgeAddress]
+		if !ok {
+			log.Debug("ignoring deposit tx from unknown bridge", "bridgeAddress", bridgeAddress)
```

Comment:
don't log in this function, just put the required error contents in the returned error, as the caller will log